### PR TITLE
Allow use of `cells_row_groups()` location in `text_transform()`

### PR DESCRIPTION
**Pull request by rich-iannone**
This change allows for the transformation of text in row groups within the `text_transform()` function.

Fixes: https://github.com/rstudio/gt/issues/894

### CODE REVIEW

File: R/dt_stub_df.R [hunk 1]
```diff
@@ -55,7 +55,7 @@ dt_stub_df_init <- function(data,
 
     # Place the `group_labels` values into `stub_df$group_id`
     stub_df[["group_id"]] <- row_group_ids
-    stub_df[["group_label"]] <- row_group_ids
+    stub_df[["group_label"]] <- lapply(row_group_ids, FUN = `[`)
```
Comment (jcheng5) [inline R/dt_stub_df.R:58]:
```suggestion
    stub_df[["group_label"]] <- as.list(row_group_ids)
```

File: R/location_methods.R [hunk 2]
```diff
@@ -331,3 +342,11 @@ to_output_location.cells_stub <- function(loc, data) {
...
   loc
 }
+
+to_output_location.cells_row_groups <- function(loc, data) {
+
+  loc <- resolve_location(loc = loc, data = data)
+
+  class(loc) <- c("output_relative", class(loc))
+  loc
+}
```
Comment (jcheng5) [inline R/location_methods.R:352]:
```suggestion
```

File: R/text_transform.R [hunk 3]
```diff
@@ -146,3 +153,35 @@ text_transform_at_location.cells_column_labels <- function(loc,
...
+
+    stub_df <- dt_stub_df_get(data = data)
+
+    if (group %in% row_group_vec) {
+
+      if (is.na(group)) next
+
+      stub_df <-
+        stub_df %>%
+        dplyr::mutate(group_label = dplyr::case_when(
```
Comment (jcheng5) [inline R/text_transform.R:177]: Let's see if we can avoid case_when so that `fn` only needs to be invoked with actual desired groups

Comment (rich-iannone) [reply R/text_transform.R]: Now fixed with this:

```r
stub_df[!is.na(stub_df$group_id) & stub_df$group_id == group, ][["group_label"]] <-
  as.list(fn(stub_df[!is.na(stub_df$group_id) & stub_df$group_id == group, ][["group_label"]]))
```

**Review by jcheng5 — APPROVED** [review]

**Pull request merged**